dynamiq/nodes/agents/react.py

```diff
@@ -587,24 +595,33 @@ def generate_structured_output_schemas(self):
         self.format_schema = schema
 
     @staticmethod
-    def filter_format_type(param_type: str) -> str:
-        "Filters proper type for an function calling schema"
-        match param_type:
-            case "bool":
-                return "boolean"
-            case "int":
-                return "integer"
-            case "float":
-                return "float"
-        return "string"
+    def filter_format_type(param_type: str | type) -> str:
+        """Filters proper type for a function calling schema."""
+        if isinstance(param_type, str):
+            match param_type:
+                case "bool":
+                    return "boolean"
+                case "int":
+                    return "integer"
+                case "float":
+                    return "float"
+                case _:
+                    return "string"
+        elif hasattr(param_type, "__origin__") and param_type.__origin__ is Union:
+            # Handle Union types by using the first type in the Union
+            first_type = param_type.__args__[0]
+            return "string" if not hasattr(first_type, "__name__") else first_type.__name__
```

Comment:
It seems that in case this condition is false value for type is not validated.

Reply:
Great, I've fixed this.